### PR TITLE
Standalone executable for gz_msgs

### PR DESCRIPTION
**Pull request by mjcarroll**
# 🎉 New feature

## Summary

Adds a C++ standalone executable for the gz-msgs command line interface.

For now this maintains the gz.hh in case someone is expecting those symbols to be there.  These should probably be considered deprecated and removed in Harmonic.

## Checklist
- [ ] Signed all commits for DCO
- [ ] Added tests
- [ ] Added example and/or tutorial
- [ ] Updated documentation (as needed)
- [ ] Updated migration guide (as needed)
- [ ] Consider updating Python bindings (if the library has them)
- [ ] `codecheck` passed (See [contributing](https://gazebosim.org/docs/all/contributing#contributing-code))
- [ ] All tests passed (See [test coverage](https://gazebosim.org/docs/all/contributing#test-coverage))
- [ ] While waiting for a review on your PR, please help review [another open pull request](https://github.com/pulls?q=is%3Aopen+is%3Apr+user%3Agazebosim+archived%3Afalse+) to support the maintainers

**Note to maintainers**: Remember to use **Squash-Merge** and edit the commit message to match the pull request summary while retaining `Signed-off-by` messages.

### CODE REVIEW

File: src/cmd/cmdmsgs.rb.in [hunk 1]
```diff
@@ -14,148 +14,52 @@
...
-    options = parse(args)
-
-    # puts 'Parsed:'
-    # puts options
+    command = args[0]
+    exe_name = COMMANDS[command]
 
-    # Read the plugin that handles the command.
-    if LIBRARY_NAME[0] == '/'
+    if exe_name[0] == '/'
```
Comment (azeey) [inline src/cmd/cmdmsgs.rb.in:33]: I don't think this works on windows. I recommend using `pathname` as
https://github.com/gazebosim/gz-sim/blob/001dd9b829eb8e8b4465d87c6c70ccf4ee2e6b6a/src/cmd/cmdsim.rb.in#L355

**Review by azeey — COMMENTED** [review]
LGTM! One small comment regarding absolute paths.

File: src/cmd/msgs_main.cc [hunk 2]
```diff
@@ -0,0 +1,98 @@
...
+     }, "List all message types.");
+
+  _app.add_option_function<std::string>("-i,--info",
+     [opt](const std::string &_msgName){
+        opt->command = MsgCommand::kMsgInfo;
+        opt->msgName = _msgName;
+     }, "Get info about the specified message type.");
+
+
+  _app.callback([opt](){
```
Comment (ahcorde) [inline src/cmd/msgs_main.cc:80]:
```suggestion
     }, "Get info about the specified message type.");

  _app.callback([opt](){
```

File: src/cmd/msgs_main.cc [hunk 3]
```diff
@@ -0,0 +1,98 @@
...
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ */
+
+#include <gz/utils/cli/CLI.hpp>
+#include <gz/utils/cli/GzFormatter.hpp>
+
```
Comment (ahcorde) [inline src/cmd/msgs_main.cc:20]:
```suggestion
#include <string>
```

**Review by ahcorde — CHANGES_REQUESTED** [review]

**Comment by mjcarroll** [comment]
After talking with @j-rivero, it makes more sense to target this to `harmonic`

**Pull request closed without merging**